src/lib/presets/index.ts

```diff
@@ -129,6 +131,66 @@ export const savePreset = async (preset: Patch) => {
   console.log('saved preset', storedPreset)
 }
 
+/**
+ * Batch write all presets before publishing to the file system
+ *
+ * @param presets Patch[]
+ */
+export const saveAllPresets = async (presets: Patch[]) => {
+  const fs = getStore(fileSystemStore)
+
+  const updatedPresets = await Promise.all(presets.map(async (preset) => {
+    const updatedPreset = {
+      ...preset,
+      creator: getUsername(),
+    }
+
+    const contentPath = webnative.path.combine(PRESETS_DIRS[updatedPreset.visibility], webnative.path.file(`${updatedPreset?.id}.json`))
+
+    await fs?.write(
+      contentPath,
+      new TextEncoder().encode(JSON.stringify(updatedPreset))
+    )
+
+    return updatedPreset
+  }))
+
+  console.log('updatedPresets', updatedPresets)
+
+  await fs?.publish()
+
+  await Promise.all(updatedPresets.map(async (preset) => {
+    presetsStore.update((state) => {
+      const updatedPresets = addOrUpdate(state.presets, preset).sort((a, b) => a.name.localeCompare(b.name, 'en', {'sensitivity': 'base'}))
+      originalPresets = JSON.parse(JSON.stringify(updatedPresets))
+
+      return {
+        ...state,
+        presets: updatedPresets,
+      }
+    })
+
+
+    // Update patchStore if it currently contains this preset
+    const patch = getStore(patchStore)
+    if (patch.id === preset?.id) {
+      patchStore.update(() => preset)
+    }
+
+    const contentPath = webnative.path.combine(PRESETS_DIRS[preset.visibility], webnative.path.file(`${preset?.id}.json`))
+
+    const storedPreset = JSON.parse(new TextDecoder().decode(
+      await fs?.read(contentPath)
+    )) as Patch
+
+    console.log('stored preset', storedPreset)
+
+    return storedPreset
+  }))
+
+  addNotification('Presets copied', 'success')
```

Comment:
We can probably remove this notification. Users won't have understand what is going on internally when we switch over their filesystem.

Reply:
removed!